infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -226,6 +231,8 @@
                         .setNameFormat(serverContext.getThreadPrefix() + "LocalMetricsPolling")
                         .build());
 
+        this.sequencerRecoveryFuture = CompletableFuture.completedFuture(true);
+
         // Creating the initialization task thread.
         // This thread pool is utilized to dispatch one time recovery and sequencer bootstrap tasks.
         // One these tasks finish successfully, they initiate the detection tasks.
```

Comment:
initializationTaskThread runs the sequencer recovery before scheduling the detector tasks, which was causing the problem in the first place, this dependency is till not fixed. Adding a handler to throw a cluster unavailable exception can throw an exception in initializationTaskThread before scheduling the detector tasks.

Reply:
The initializationTaskThread will eventually call runLayoutReconfiguration which calls reconfigureSequencerServers. That method can block forever or throw an exception in either case the fault detectors wont be scheduled. That is one of the main problems and I don't see how this PR addresses that issue.

Reply:
The initializationTask does perform sequencer recovery but they are not blocking forever. They will now bail out with a clusterUnreachableException. These exceptions thrown by the sequener recovery task are logged and since they are best effort tasks, they do not affect the detector tasks.